frontend/src/components/panels/Document.vue

```diff
@@ -457,6 +499,19 @@ export default defineComponent({
 		});
 		this.editor.dispatcher.subscribeJsMessage(TriggerViewportResize, this.viewportResize);
 
+		this.editor.dispatcher.subscribeJsMessage(UpdateImageData, (updateImageData) => {
+			updateImageData.image_data.forEach((element) => {
+				// Using updateImageData.image_data.buffer returns undefined for some reason?
+				const blob = new Blob([new Uint8Array(element.image_data.values()).buffer], { type: element.mime });
+
+				const url = URL.createObjectURL(blob);
+
+				createImageBitmap(blob).then((i) => {
```

Comment:
Nitpick: `img` or `image` would avoid any possibility of confusion with the `i`ndex variable usually named `i`.